sklearn/semi_supervised/_self_training.py

```diff
@@ -238,6 +255,12 @@ def fit(self, X, y):
                 UserWarning,
             )
 
+        if _routing_enabled():
+            routed_params = process_routing(self, "fit", **fit_params)
+        else:
+            routed_params = Bunch()
+            routed_params.base_estimator = Bunch(fit=fit_params)
```

Comment:
if metadata routing is not enabled, we raise when `params` are passed anyway, so this is always empty.
```suggestion
            routed_params = Bunch()
            routed_params.base_estimator = Bunch(fit={})
```